File: src/providers/mock.ts
import type { OverlaySnapshot } from "../overlay/types";

const hhmm = (minutes: number): string => {
  const hours = Math.floor(minutes / 60);
  const mins = minutes % 60;
  return `${String(hours).padStart(2, "0")}:${String(mins).padStart(2, "0")}`;
};

export const getMockSnapshot = (): OverlaySnapshot => {
  const now = Date.now();
  const tick = Math.floor(now / 1000);

  const claudeRemaining = 72 - (tick % 11);
  const codexRemaining = 58 - (tick % 7);

  const toStatus = (remaining: number) => {
    if (remaining <= 20) {
      return "critical" as const;
    }
    if (remaining <= 45) {
      return "warn" as const;
    }
    return "ok" as const;
  };

  return {
    title: "quota hud",
    mode: "mock overlay",
    updatedAt: new Date(now).toLocaleTimeString("zh-CN", {
      hour: "2-digit",
      minute: "2-digit",
      second: "2-digit",
      hour12: false,
    }),
    providers: [
      {
        id: "claude",
        label: "claude",
        remainingLabel: `${claudeRemaining}%`,
        remainingPercent: claudeRemaining,
        windowLabel: "5h 窗口",
        resetInLabel: hhmm(102),
        detail: "剩余会话额度",
        status: toStatus(claudeRemaining),
      },
      {
        id: "codex",
        label: "codex",
        remainingLabel: `${codexRemaining}%`,
        remainingPercent: codexRemaining,
        windowLabel: "周额度",
        resetInLabel: "2d 04h",
        detail: "剩余本周额度",
        status: toStatus(codexRemaining),
      },
    ],
  };
};

File: src-tauri/src/providers/mock.rs
use std::time::{Duration, SystemTime, UNIX_EPOCH};

use super::{OverlaySnapshot, ProviderSnapshot};

fn now_seconds() -> u64 {
    SystemTime::now()
        .duration_since(UNIX_EPOCH)
        .unwrap_or(Duration::from_secs(0))
        .as_secs()
}

fn hhmm(total_minutes: u64) -> String {
    let hours = total_minutes / 60;
    let minutes = total_minutes % 60;
    format!("{hours:02}:{minutes:02}")
}

fn status_for(remaining: u8) -> &'static str {
    if remaining <= 20 {
        "critical"
    } else if remaining <= 45 {
        "warn"
    } else {
        "ok"
    }
}

pub fn mock_overlay_snapshot() -> OverlaySnapshot {
    let tick = now_seconds();
    let claude_remaining = 72u8.saturating_sub((tick % 11) as u8);
    let codex_remaining = 58u8.saturating_sub((tick % 7) as u8);

    OverlaySnapshot {
        title: "quota hud".to_string(),
        mode: "tauri mock".to_string(),
        updated_at: format!("t+{}", tick % 100_000),
        providers: vec![
            ProviderSnapshot {
                id: "claude".to_string(),
                label: "claude".to_string(),
                remaining_label: format!("{claude_remaining}%"),
                remaining_percent: claude_remaining,
                window_label: "5h 窗口".to_string(),
                reset_in_label: hhmm(102),
                detail: "剩余会话额度".to_string(),
                status: status_for(claude_remaining).to_string(),
            },
            ProviderSnapshot {
                id: "codex".to_string(),
                label: "codex".to_string(),
                remaining_label: format!("{codex_remaining}%"),
                remaining_percent: codex_remaining,
                window_label: "周额度".to_string(),
                reset_in_label: "2d 04h".to_string(),
                detail: "剩余本周额度".to_string(),
                status: status_for(codex_remaining).to_string(),
            },
        ],
    }
}
